src/fits.cc

```diff
@@ -1316,14 +1316,12 @@ void Fits::writeImage(image::ImageBase<T> const &image, ImageWriteOptions const
     // Write the header
     std::shared_ptr<daf::base::PropertyList> wcsMetadata =
             geom::createTrivialWcsMetadata(image::detail::wcsNameForXY0, image.getXY0());
+    std::shared_ptr<daf::base::PropertySet> fullMetadata(wcsMetadata);
     if (header) {
-        std::shared_ptr<daf::base::PropertySet> copy = header->deepCopy();
-        copy->combine(*wcsMetadata);
-        header = copy;
-    } else {
-        header = wcsMetadata;
+        fullMetadata = header->deepCopy();
+        fullMetadata->combine(*wcsMetadata);
     }
```

Comment:
I appreciate the desire to avoid the confusing overwrite of `header`, but it's also confusing to initialize `fullMetadata` with one value and then immediately replace it. Perhaps:
```suggestion
    std::shared_ptr<daf::base::PropertySet> fullMetadata;
    if (header) {
        fullMetadata = header->deepCopy();
        fullMetadata->combine(*wcsMetadata);
    } else {
        fullMetadata = wcsMetadata;
    }
```